creator/utils.py

```diff
@@ -0,0 +1,85 @@
+import time
+import logging
+from rq.job import NoSuchJobError, JobStatus
+from rq.command import send_kill_horse_command
+from rq.worker import Worker, WorkerStatus
+
+WAIT_FOR_FAIL_TIMEOUT = 30
+
+
+def stop_job(job_id, queue=None, delete=True, logger=None):
+    """
+    Stop a running rq job and optionally delete the job from the rq queue
+    and registries.
+
+    NOTE: This method may not be needed after this
+    https://github.com/rq/rq/issues/1371 is implemented
+
+    For jobs that have failed, do nothing
+    For jobs that are queued or deferred cancel or delete them (see rq.job.Job)
+    For jobs that have started, kill the worker horse executing the job
+
+    If delete is True, remove the job from the redis queue, all rq registries,
+    and from redis
+
+    If delete is False just remove the job from the redis queue
+
+    :param job_id: The redis job id
+    :type job_id: str
+    :param queue: The queue the job was submitted to
+    :type queue: rq.Queue
+    :param delete: whether to delete or cancel the job
+    :type delete: bool
+    """
+    if not logger:
+        logger = logging.getLogger(__name__)
+
+    logger.info(f"Preparing to stop redis job {job_id}")
+
+    # Find rq job
+    if not queue:
+        queue = Queue(connection=Redis())
```

Comment:
Where does `Redis` come from?
I think `django_rq.get_queue()` may be preferable here.

Reply:
I switched it from `django_rq` bc I thought about submitting a PR to rq's repo but I don't think I'm going to do that anymore so I'll switch it back